File: app/src/main/java/br/com/alura/leilao/api/retrofit/converter/StringConverterFactory.java
package br.com.alura.leilao.api.retrofit.converter;

import androidx.annotation.Nullable;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;
import retrofit2.Retrofit;

public class StringConverterFactory extends Converter.Factory{

   private static final MediaType MEDIA_TYPE = MediaType.parse("text/plain");

   @Nullable
   @Override
   public Converter<?, RequestBody> requestBodyConverter(
      Type type, Annotation[] parameterAnnotations, Annotation[] methodAnnotations, Retrofit retrofit)
   {
      if(String.class.equals(type)){
         return new Converter<String, RequestBody>(){
            @Override
            public RequestBody convert(String value){
               return RequestBody.create(MEDIA_TYPE, value);
            }
         };
      }
      return null;
   }
}

File: app/src/main/java/br/com/alura/leilao/api/retrofit/converter/StringConverterFactory.kt
package br.com.alura.leilao.api.retrofit.converter

import okhttp3.MediaType
import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.RequestBody
import retrofit2.Converter
import retrofit2.Retrofit
import java.lang.reflect.Type

class StringConverterFactory : Converter.Factory() {

    private val MEDIA_TYPE = "text/plain".toMediaTypeOrNull()

    override fun requestBodyConverter(
        type: Type,
        parameterAnnotations: Array<Annotation>,
        methodAnnotations: Array<Annotation>,
        retrofit: Retrofit
    ): Converter<*, RequestBody>? {
        return if (String::class.java == type) {
            Converter<String, RequestBody> { value ->
                RequestBody.create(MEDIA_TYPE, value)
            }
        } else {
            null
        }
    }
}